internal/domain/file_content_response.go

```diff
@@ -0,0 +1,9 @@
+package domain
+
+type FileContentResponse struct {
+	Filename  string `json:"filename"`
+	Content   string `json:"content"`
+	StartLine int    `json:"start_line"`
```

Comment:
Consider adding validation for the StartLine and EndLine fields to ensure that StartLine is less than or equal to EndLine. This will prevent logical errors when these values are used, particularly in scenarios where they are expected to represent a range.